feat: add CSP rules to nextjs base configuration

nikomakela opened the pull request:
HH-432 HH-433.

Set CSP rules (copied from [Kultus
UI](https://github.com/City-of-Helsinki/palvelutarjotin-ui/blob/main/next.config.js#L172-L184) ). Improved the syntax a bit to make it more configurable. Also checked that the API CORS are set OK.

nikomakela committed:
feat: add CSP rules to nextjs base configuration

HH-432 HH-433.

Set CSP rules (copied from [Kultus
UI](https://github.com/City-of-Helsinki/palvelutarjotin-ui/blob/main/next.config.js#L172-L184)
). Improved the syntax a bit to make it more configurable. Also checked
that the API CORS are set OK.

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr795.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr795.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
fix(apollo-client): add variables to error link log entry

HH-432.

Without variables, it's imporssible to know, which query actually
failed.

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr795.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr795.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr795.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
feat(router): add cors rules to apollo federation router

HH-433.

Add CORS rules to Apollo Federation Router.

The Headless CMS has a previewing feature, which is why it needs to be
added in CORS rules. That feature also needs authorization headers.

Otherwise, `localhost:*` is for local development and `*.hel.ninja` and
`*.hel.fi` are for City of Helsinki addresses.

terovirtanen commented:
**Federation Router branch is deployed to platta: https://events-graphql-federation-events-pr795.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr795.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr795.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Federation Router branch is deployed to platta: https://events-graphql-federation-events-pr795.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Federation Router branch is deployed to platta: https://events-graphql-federation-sports-pr795.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr795.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr795.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr795.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders reviewed (approved):
As CSP is applied also to the page routes it's ok, but out of curiosity what was the reason for adding `X-XSS-Protection = 0` to the page routes?

nikomakela commented:
> As CSP is applied also to the page routes it's ok, but out of curiosity what was the reason for adding `X-XSS-Protection = 0` to the page routes?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/X-XSS-Protection
<img width="836" height="826" alt="image" src="https://github.com/user-attachments/assets/a491798c-630f-443c-bbef-1370902a1771" />
